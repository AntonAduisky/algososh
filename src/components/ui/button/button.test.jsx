import React from "react";
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from "@testing-library/react";

import { Button } from "./button";

describe("Check props button", () => {
  it("Button props text", () => {
    const tree = renderer.create(<Button text={"Test"} />).toJSON();
    expect(tree).toMatchSnapshot(); // для генерации снепшота
  });

  it("button no text", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot(); // для генерации снепшота
  });

  it("button disabled", () => {
    const tree = renderer.create(<Button disabled />).toJSON();
    expect(tree).toMatchSnapshot(); // для генерации снепшота
  });

  it("button isLoading", () => {
    const tree = renderer.create(<Button isLoader />).toJSON();
    expect(tree).toMatchSnapshot(); // для генерации снепшота
  });
  
  it('button callback', () => {
    
    window.alert = jest.fn();

        // Рендерим компонент
    render(<Button onClick={()=> window.alert('text')} text={'test text'}/>)

        // Находим элемент ссылки
      // eslint-disable-next-line no-restricted-globals
      const button = screen.getByText('test text');

        // Имитируем нажатие на ссылку
    fireEvent.click(button);
        
        // Проверяем, что alert сработал с правильным текстом предупреждения
    expect(window.alert).toHaveBeenCalledWith('text');
})
});
