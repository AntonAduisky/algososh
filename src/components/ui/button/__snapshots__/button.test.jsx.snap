// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check props button Button props text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Test
  </p>
</button>
`;

exports[`Check props button button disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Check props button button isLoading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Check props button button no text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
